latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,210 @@
+                    "run Rscript /opt/latch/environment.R",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            )
+        )
+
+    conda_env_file_name = None
+    if (pkg_root / "environment.yml").exists():
+        conda_env_file_name = "environment.yml"
+    elif (pkg_root / "environment.yaml").exists():
+        conda_env_file_name = "environment.yaml"
+
+    if conda_env_file_name is not None:
+        print("Adding conda + conda environment installation phase")
+        with (pkg_root / conda_env_file_name).open("rb") as f:
+            conda_env = yaml.safe_load(f)
+
+        if "name" in conda_env:
+            env_name = conda_env["name"]
+        else:
+            env_name = "unnamed"
+
+        commands += [
+            DockerCmdBlock(
+                comment="set conda environment variables",
+                commands=[
+                    "env CONDA_DIR /opt/conda",
+                    "env PATH=$CONDA_DIR/bin:$PATH",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            ),
+            DockerCmdBlock(
+                comment="install miniconda",
+                commands=[
+                    dedent(
+                        r"""
+                        run apt-get update --yes && \
+                            apt-get install --yes curl && \
+                            curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
+                            mkdir /root/.conda && \
+                            bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda && \
+                            rm -f Miniconda3-latest-Linux-x86_64.sh && \
+                            conda init bash
+                        """
+                    ).strip(),
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            ),
+            DockerCmdBlock(
+                comment="build and configure conda environment",
+                commands=[
+                    f"copy {conda_env_file_name} /opt/latch/{conda_env_file_name}",
+                    f"run conda env create --file /opt/latch/{conda_env_file_name} --name {env_name}",
+                    f"""shell ["conda", "run", "--name", "{env_name}", "/bin/bash", "-c"]""",
+                    "run /opt/conda/bin/pip install --upgrade latch",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            ),
+        ]
+
+    if (pkg_root / "setup.py").exists() or (pkg_root / "pyproject.toml").exists():
+        print("Adding local package installation phase")
+        commands.append(
+            DockerCmdBlock(
+                comment="add local package to python environment",
+                commands=["run pip install --editable /root/"],
+                order=DockerCmdBlockOrder.postcopy,
+            )
+        )
+
+    if (pkg_root / "requirements.txt").exists():
+        print("Adding python package installation phase")
```

Comment:
```suggestion
        print("Adding python dependency installation phase")
```
otherwise sounds the same as the "local package" installation phase